src/util/BitField.js

```diff
@@ -133,18 +141,20 @@ class BitField {
    * * A bit number (this can be a number literal or a value taken from {@link BitField.FLAGS})
    * * An instance of BitField
    * * An Array of BitFieldResolvable
-   * @typedef {number|BitField|BitFieldResolvable[]} BitFieldResolvable
+   * @typedef {number|bigint|BitField|BitFieldResolvable[]} BitFieldResolvable
    */
 
   /**
    * Resolves bitfields to their numeric form.
-   * @param {BitFieldResolvable} [bit=0] - bit(s) to resolve
-   * @returns {number}
+   * @param {BitFieldResolvable} [bit] - bit(s) to resolve
+   * @returns {number|bigint}
    */
-  static resolve(bit = 0) {
-    if (typeof bit === 'number' && bit >= 0) return bit;
+  static resolve(bit) {
+    const defaultBit = this.name === 'Permissions' ? 0n : 0;
+    if (typeof bit === 'undefined') return defaultBit;
+    if (['number', 'bigint'].includes(typeof bit) && bit >= defaultBit) return bit;
```

Comment:
```suggestion
    if (typeof defaultBit === typeof bit && bit >= defaultBit) return bit;
```

Otherwise `new Permissions(0)` (or bigints for other bitfields) would be allowed and further calls would throw an error:

e.g. 
```js
new Permission(0).add("ADMINISTRATOR")
// TypeError: Cannot mix BigInt and other types, use explicit conversions
//     at Permissions.add (.../BitField.js:84:33)
```